gematria/experiments/access_pattern_bm/linked_list.cc

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gematria/experiments/access_pattern_bm/linked_list.h"
+
+#include <immintrin.h>
+
+#include <memory>
+#include <random>
+
+namespace gematria {
+
+static std::default_random_engine generator;
+static std::uniform_int_distribution<int> distribution(0, 1023);
+
+std::unique_ptr<Node, void (*)(Node *)> CreateRandomLinkedList(
+    const std::size_t size) {
+  Node *head = new Node;
+  Node *current = head;
+
+  int i = size;
+  while (i--) {
+    current->next = new Node;
+    current->value = distribution(generator);
+    current = current->next;
+  }
+
+  auto deleter = [](Node *ptr) {
+    while (ptr) {
+      Node *temp = ptr->next;
+      delete ptr;
+      ptr = temp;
+    }
+  };
```

Comment:
Optional: You could define this as a functor
```
struct NodeDeleter {
  void operator()(Node* node) {
    while (ptr) {
      Node *temp = ptr->next;
      delete ptr;
      ptr = temp;
    }
  }
};
```
and then use it without having to instantiate the deleter explicitly, e.g. as
```
return std::unique_ptr<Node, NodeDeleter>(head);
```

It has the additional advantage that NodeDeleter is empty, so the unique_ptr<> doesn't need to carry a function pointer with it (i.e. `sizeof(std::unique_ptr<Node, NodeDeleter>) == 8` while (`sizeof(std::unique_ptr<Node, void(*)(Node*)>) == 16`. It doesn't really matter here though.